### PR TITLE
feat: loosen react deps

### LINKED ISSUE #851: React 18 support?
We are not able to add this in Next.js 14. Can you add support for react 18?

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] | fix #851 |
| :--------------------: | -------- |

## 🧰 Changes

Loosens the react peer deps so it won't complain when installing with react > 16.

This runs the visual tests in only one environment. I don't see a lot of value in running them against all versions?

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

**Review by rafegoldberg — APPROVED** [review]
looks okay to me

**Comment by erunion** [comment]
Are there any tests running against React 18?

**Comment by rafegoldberg** [comment]
@erunion I don't think so

**Comment by rafegoldberg** [comment]
…that is a very good point you make

**Comment by rafegoldberg** [comment]
welp

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,7 +36,8 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [18.x, 20.x, 21.x]
+        node-version: [21.x]
```
Comment (erunion) [inline .github/workflows/ci.yml:39]: would recommend running these against node 20 as it's an LTS release and less prone to issues hitting their release channel

```suggestion
        node-version: [20.x]
```

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ARG NODE_VERSION=18
-FROM node:${NODE_VERSION}-buster
+FROM node:${NODE_VERSION}
```
Comment (erunion) [inline Dockerfile:2]: might want to pull in an alpine release here so the docker container isn't massive

```suggestion
FROM node:${NODE_VERSION}-alpine3.18
```

**Review by erunion — APPROVED** [review]

**Review by rafegoldberg — APPROVED** [review]
+1 to Jon's [rec], otherwise this seems good to me! 👌

[rec]: https://github.com/readmeio/markdown/pull/852/files#r1556589018

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.78.0</b>](/readmeio/markdown/releases/tag/v6.78.0)
<br>

**Pull request merged**